src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services10.php

```diff
@@ -80,10 +80,12 @@ protected function getTestService()
      */
     public function getParameter($name)
     {
-        $name = strtolower($name);
+        if (!(isset($this->parameters[$name]) || isset($this->loadedDynamicParameters[$name]) || array_key_exists($name, $this->parameters))) {
+            $name = strtolower($name);
```

Comment:
+1,000 !!!  We should stop changing the user config "magically" behind the scenes ... and that includes service IDs, param IDs, etc. ... unless there are strong technical reasons to do so. Thanks!